msvc builds cause warning `c4530`

### Godot version

v4.2.beta.custom_build [5ee983188]

### godot-cpp version

c1196a1ab0a1ca166d0e5e2f08f9fe4156118c5e

### System information

Windows 11

### Issue description

Recently, my godot-cpp builds have had their logs absolutely FLOODED with this message en-masse:
```
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\vector(1612): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\vector(1601): note: while compiling class template member function 'void std::vector<GDExtensionClassMethodArgumentMetadata,std::allocator<GDExtensionClassMethodArgumentMetadata>>::_Reallocate_exactly(const unsigned __int64)'
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\vector(1685): note: see the first reference to 'std::vector<GDExtensionClassMethodArgumentMetadata,std::allocator<GDExtensionClassMethodArgumentMetadata>>::_Reallocate_exactly' in 'std::vector<GDExtensionClassMethodArgumentMetadata,std::allocator<GDExtensionClassMethodArgumentMetadata>>::reserve'
D:\Godot\godot-libusb\godot-cpp\include\godot_cpp/core/method_bind.hpp(134): note: see the first reference to 'std::vector<GDExtensionClassMethodArgumentMetadata,std::allocator<GDExtensionClassMethodArgumentMetadata>>::reserve' in 'godot::MethodBind::get_arguments_metadata_list'
D:\Godot\godot-libusb\godot-cpp\include\godot_cpp/core/method_bind.hpp(131): note: see reference to class template instantiation 'std::vector<GDExtensionClassMethodArgumentMetadata,std::allocator<GDExtensionClassMethodArgumentMetadata>>' being compiled
```
I haven't been actively building a gdextension up until this past week or so, so I can't speak for how long this has potentially been in place. The last time I was actively building was late July, and it wasn't present at that time

### Steps to reproduce

1. build with msvc
2. cry

### Minimal reproduction project

N/A

--- a/tools/godotcpp.py
+++ b/tools/godotcpp.py
@@ -179,7 +179,7 @@ def options(opts, env):
         BoolVariable(
             key="use_hot_reload",
             help="Enable the extra accounting required to support hot reload.",
-            default=(env.get("target", "template_debug") != "template_release"),
+            default=env.get("use_hot_reload", None),
         )
     )
 
@@ -245,8 +245,19 @@ def generate(env):
 
     print("Building for architecture " + env["arch"] + " on platform " + env["platform"])
 
+    if env.get("use_hot_reload") is None:
+        env["use_hot_reload"] = env["target"] != "template_release"
     if env["use_hot_reload"]:
         env.Append(CPPDEFINES=["HOT_RELOAD_ENABLED"])
+
+    tool = Tool(env["platform"], toolpath=["tools"])
+
+    if tool is None or not tool.exists(env):
+        raise ValueError("Required toolchain not found for platform " + env["platform"])
+
+    tool.generate(env)
+    target_tool = Tool("targets", toolpath=["tools"])
+    target_tool.generate(env)
 
     # Disable exception handling. Godot doesn't use exceptions anywhere, and this
     # saves around 20% of binary size and very significant build time.
@@ -257,15 +268,6 @@ def generate(env):
             env.Append(CXXFLAGS=["-fno-exceptions"])
     elif env.get("is_msvc", False):
         env.Append(CXXFLAGS=["/EHsc"])
-
-    tool = Tool(env["platform"], toolpath=["tools"])
-
-    if tool is None or not tool.exists(env):
-        raise ValueError("Required toolchain not found for platform " + env["platform"])
-
-    tool.generate(env)
-    target_tool = Tool("targets", toolpath=["tools"])
-    target_tool.generate(env)
 
     # Require C++17
     if env.get("is_msvc", False):

